File: js-mocha/test/test_gilded_rose.js
var { expect } = require("chai");
var { Shop, Item } = require("../src/gilded_rose.js");
describe("Gilded Rose", function () {
  describe("updateQuality", function () {
    it("should return foo", function () {
      const gildedRose = new Shop([new Item("foo", 0, 0)]);
      const items = gildedRose.updateQuality();
      expect(items[0].name).to.equal("foo");
    });

    it("should reduce sell in date by 1", function () {
      const gildedRose = new Shop([new Item("foo", 2, 0)]);
      const items = gildedRose.updateQuality();
      expect(items[0].sellIn).to.equal(1);
    });

    it("should reduce quality by 1", function () {
      const gildedRose = new Shop([new Item("foo", 2, 3)]);
      const items = gildedRose.updateQuality();
      expect(items[0].quality).to.equal(2);
    });

    it("should reduce quality by 2 if sell-in date is 0", function () {
      const gildedRose = new Shop([new Item("foo", 0, 10)]);
      const items = gildedRose.updateQuality();
      expect(items[0].quality).to.equal(8);
    });

    it("should not set quality below 0", function () {
      const gildedRose = new Shop([new Item("foo", 3, 0)]);
      const items = gildedRose.updateQuality();
      expect(items[0].quality).to.equal(0);
    });
  });
});

File: python/test_gilded_rose.py
# -*- coding: utf-8 -*-
import unittest

from gilded_rose import Item, GildedRose


class GildedRoseTest(unittest.TestCase):
    def test_foo(self):
        items = [Item("foo", 0, 0)]
        gilded_rose = GildedRose(items)
        gilded_rose.update_quality()
        self.assertEquals("foo", items[0].name)

    def test_item_sell_in_decreases(self):
        items = [Item("foo", 0, 0)]
        gilded_rose = GildedRose(items)
        gilded_rose.update_quality()
        self.assertEquals(-1, items[0].sell_in)

        
if __name__ == '__main__':
    unittest.main()
